src/modules/users-permissions/PermissionManager.js

```diff
@@ -0,0 +1,35 @@
+
+class PermissionManager {
+  constructor(user, permissionObjs) {
+    this._user = user
+    this._permisionObjs = permissionObjs
+    
+    // construct user permissions list
+    this._permissions = new Set()
+    this._user.roles.forEach((role) => {
+      for (let p of role.permissions) {
+        this._permissions.add(p)
+      }
+    })
+  }
+
+  allow(permissionName, obj = null) {
```

Comment:
Ah yep. The proposal in #20 was to name this `allowed` (`req.user.permissions.allowed()` now, does that still sound too close? (As discussed there, not a good idea for this runtime call to support the same permissions-registering intent.)